extensions/resource-deployment/src/services/tools/azdataTool.ts

```diff
@@ -53,7 +56,67 @@ export class AzdataTool extends ToolBase {
 		return true;
 	}
 
-	public install(): Promise<void> {
-		return Promise.reject('not implemented');
+	protected async getInstallationPath(): Promise<string | undefined> {
+		switch (this.osType) {
+			case OsType.linux:
+				return installationRoot;
+			default:
+				return path.join(await this.getPip3InstallLocation('azdata-cli'), '..', 'Scripts');
+		}
 	}
+
+	readonly allInstallationCommands: Map<OsType, Command[]> = new Map<OsType, Command[]>([
+		[OsType.linux, linuxInstallationCommands],
+		[OsType.win32, defaultInstallationCommands],
+		[OsType.darwin, defaultInstallationCommands],
+		[OsType.others, defaultInstallationCommands]
+	]);
 }
+
+const linuxInstallationCommands = [
+	{
+		sudo: true,
+		comment: localize('resourceDeployment.Azdata.AptGetUpdate', 'updating repository information before installing azdata ...'),
+		command: 'apt-get update'
+	},
+	{
+		sudo: true,
+		comment: localize('resourceDeployment.Azdata.AptGetPackages', 'getting packages needed for azdata installation ...'),
+		command: 'apt-get install gnupg ca-certificates curl apt-transport-https lsb-release -y'
+	},
+	{
+		sudo: true,
+		comment: localize('resourceDeployment.Azdata.DownloadAndInstallingSigningKey', 'downloading and installing the signing key for azdata ...'),
+		command: 'wget -qO- https://packages.microsoft.com/keys/microsoft.asc | apt-key add -'
+	},
+	{
+		sudo: true,
+		comment: localize('resourceDeployment.Azdata.AddingAzureCliRepositoryInformation', 'adding the azdata repository information ...'),
+		command: 'add-apt-repository "$(wget -qO- https://packages.microsoft.com/config/ubuntu/16.04/mssql-server-preview.list)"'
+	},
+	{
+		sudo: true,
+		comment: localize('resourceDeployment.Azdata.AptGetUpdateAgain', 'updating repository information again for azdata ...'),
+		command: 'apt-get update'
+	},
+	{
+		sudo: true,
+		comment: localize('resourceDeployment.Azdata.InstallingAzdata', 'installing azdata ...'),
+		command: 'apt-get install -y azdata-cli'
+	}
+];
+
+const defaultInstallationCommands = [
+	{
```

Comment:
please remove the uninstall logic, we can assume it is a clean start.

Reply:
but what is this harm and it is already tested to work alright.

Reply:
done